### PR TITLE
Create Makefile

### PR DESCRIPTION
**Pull request by RONTheCookie**
This PR adds a makefile which is a pretty standard way for programs to be built and installed in GNU/Linux land.
Here is the process I went through to build and install Ion using my Makefile:
```shell
ron@cookiemonster ~/git/Ion -> make 
dotnet build -c Release
Microsoft (R) Build Engine version 15.9.20.63311 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restoring packages for /home/ron/git/Ion/Ion/Ion.csproj...
/home/ron/git/Ion/Ion/Ion.csproj : warning NU1701: Package 'LibLLVM 5.0.1' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.2'. This package may not be fully compatible with your project. [/home/ron/git/Ion/ion.sln]
  Generating MSBuild file /home/ron/git/Ion/Ion/obj/Ion.csproj.nuget.g.props.
  Generating MSBuild file /home/ron/git/Ion/Ion/obj/Ion.csproj.nuget.g.targets.
  Restore completed in 263.33 ms for /home/ron/git/Ion/Ion/Ion.csproj.
/home/ron/git/Ion/Ion/Ion.csproj : warning NU1701: Package 'LibLLVM 5.0.1' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.2'. This package may not be fully compatible with your project.
  Ion -> /home/ron/git/Ion/Ion/bin/Release/netcoreapp2.2/Ion.dll

Build succeeded.

/home/ron/git/Ion/Ion/Ion.csproj : warning NU1701: Package 'LibLLVM 5.0.1' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.2'. This package may not be fully compatible with your project. [/home/ron/git/Ion/ion.sln]
/home/ron/git/Ion/Ion/Ion.csproj : warning NU1701: Package 'LibLLVM 5.0.1' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.2'. This package may not be fully compatible with your project.
    2 Warning(s)
    0 Error(s)

Time Elapsed 00:00:02.42
ron@cookiemonster ~/git/Ion -> sudo make install
# TODO: Make this path not constant 
mkdir /opt/Ion
cp -r Ion/bin/Release/netcoreapp2.2 /opt/Ion
ron@cookiemonster ~/git/Ion -> ls /opt/Ion/netcoreapp2.2/
Ion.deps.json  Ion.dll  Ion.pdb
```

### CODE REVIEW

**Comment by pi-pi3** [comment]
`/opt/Ion` isn't present in `PATH` by default. Could you add something like `ln -s /opt/Ion/ion /usr/bin` to the `install` rule?

**Comment by RONTheCookie** [comment]
> `/opt/Ion` isn't present in `PATH` by default. Could you add something like `ln -s /opt/Ion/ion /usr/bin` to the `install` rule?

That is what I intend to do in my next PR which will be for Ion.CLI.

**Comment by cloudrex** [comment]
We need to address the following concerns.

1. .NET Core package is required in the user's machine.
2. Building from source conflicts with Ion's fully managed core principle.

Maybe this should be an option, but I doubt it would be the standard method for Ion's installation since it infringes a core principle. Even if this is a common way for Linux users to install software, there are many other, more convinient ways (for the users).

However, having this is great for Linux users who wish to build from source. The more options the better.

**Comment by cloudrex** [comment]
Note that instead of making more installation instructions, you could simply invokd the `INSTALL.sh` under Ion.CLI's `.installers`, if it does not do what you intend, you are free to improve it. This way we avoid fragmentation.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+all: Ion
+	dotnet build -c Release
+
+clean:
+	rm -r Ion/bin Ion/obj
+
+install:
+	# TODO: Make this path not constant 
+	mkdir /opt/Ion
+	cp -r Ion/bin/Release/netcoreapp2.2 /opt/Ion
```
Comment (cloudrex) [inline Makefile]: Note that within the `netcoreapp2.2` there may be another folder with the platform name, so you might be copying a folder without knowing. Not entirely sure in this case however.

**Pull request merged**